src/coffee/export.coffee

```diff
@@ -99,15 +108,118 @@ class Export
     new ExportMapping(@options)
 
   # return the correct product service in case query string is used or not
-  _getProductService: (staged = true) ->
+  _getProductService: (staged = true, customCondition = false) ->
     productsService = @client.productProjections
+    if customCondition
+      productsService.where(customCondition)
+
     if @queryOptions.queryString
       productsService.byQueryString(@queryOptions.queryString, @queryOptions.isQueryEncoded)
       productsService
     else
       productsService.all().perPage(500).staged(staged)
 
-  export: (templateContent, outputFile, staged = true) =>
+  _fetchResources: =>
+    data = [
+      @typesService.getAll @client
+      @categoryService.getAll @client
+      @channelService.getAll @client
+      @customerGroupService.getAll @client
+      @taxService.getAll @client
+    ]
+    Promise.all(data)
+    .then ([productTypes, categories, channels, customerGroups, taxes]) =>
+      @typesService.buildMaps productTypes.body.results
+      @categoryService.buildMaps categories.body.results
+      @channelService.buildMaps channels.body.results
+      @customerGroupService.buildMaps customerGroups.body.results
+      @taxService.buildMaps taxes.body.results
+
+      console.warn "Fetched #{productTypes.body.total} product type(s)."
+      Promise.resolve({productTypes, categories, channels, customerGroups, taxes})
+
+  exportDefault: (templateContent, outputFile, staged = true) =>
+    @_fetchResources()
+    .then ({productTypes}) =>
+      @export templateContent, outputFile, productTypes, staged, false, true
+
+  _archiveFolder: (inputFolder, outputFile) ->
+    output = fs.createWriteStream(outputFile)
+    archive = archiver 'zip'
+
+    new Promise (resolve, reject) ->
+      output.on 'close', () -> resolve()
+      archive.on 'error', (err) -> reject(err)
+      archive.pipe output
+
+      archive.bulk([
+        { expand: true, cwd: inputFolder, src: ['**'], dest: 'products'}
+      ])
+      archive.finalize()
+
+  exportFull: (output, staged = true) =>
+    lang = GLOBALS.DEFAULT_LANGUAGE
+    console.log 'Creating full export for "%s" language', lang
+
+    @_fetchResources()
+    .then ({productTypes}) =>
+      tempDir = tmp.dirSync({ unsafeCleanup: true })
+
+      if productTypes.body.results.length
```

Comment:
I will do that .. the only thing I am concerned about is how Impex will handle if the output file will not exist.